controlplane/kubeadm/controllers/controller.go

```diff
@@ -153,6 +154,14 @@ func (r *KubeadmControlPlaneReconciler) Reconcile(req ctrl.Request) (res ctrl.Re
 			reterr = kerrors.NewAggregate([]error{reterr, err})
 		}
 
+		// TODO: remove this as soon as we have a proper remote cluster cache in place.
+		// if we are not going in exponential backoff due to error, if we are not already re-queueing, if the object is not deleted
```

Comment:
```suggestion
		// Make KCP to requeue in case status is not ready, so we can check for node status without waiting for a full resync (by default 10 minutes).
		// Only requeue if we are not going in exponential backoff due to error, or if we are not already re-queueing, or if the object has a deletion timestamp.
```